File: src/ros2_cpp_tutorial/src/custom_msg_publisher.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "ros2_interfaces_tutorial/msg/sphere.hpp"

using namespace std::chrono_literals;

class CustomMsgPublisher : public rclcpp::Node
{
  public:
    CustomMsgPublisher()
    : Node("custom_msg_publisher")
    {
      publisher_ = this->create_publisher<ros2_interfaces_tutorial::msg::Sphere>("topic", 10);
      timer_ = this->create_wall_timer(
      500ms, std::bind(&CustomMsgPublisher::timer_callback, this));
    }

  private:
    void timer_callback()
    {
      auto message = ros2_interfaces_tutorial::msg::Sphere();
      message.center.x = 1;
      message.center.y = 2;
      message.center.z = 3;
      publisher_->publish(message);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<ros2_interfaces_tutorial::msg::Sphere>::SharedPtr publisher_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<CustomMsgPublisher>());
  rclcpp::shutdown();
  return 0;
}

File: src/ros2_python_tutorial/ros2_python_tutorial/custom_msg_publisher.py
import rclpy
from rclpy.node import Node

from ros2_interfaces_tutorial.msg import Sphere

class CustomMsgPublisher(Node):

    def __init__(self):
        super().__init__('custom_msg_publisher')
        self.publisher_ = self.create_publisher(Sphere, 'topic', 10)
        timer_period = 0.5  # seconds
        self.timer = self.create_timer(timer_period, self.timer_callback)

    def timer_callback(self):
        msg = Sphere()
        msg.center.x = 1.0
        msg.center.y = 2.0
        msg.center.z = 3.0
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    minimal_publisher = CustomMsgPublisher()

    rclpy.spin(minimal_publisher)

    # Destroy the node explicitly
    # (optional - otherwise it will be done automatically
    # when the garbage collector destroys the node object)
    minimal_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()